### DISCUSSION TITLE
GloFAS migrated to EWDS Portal, version 3.1 will not be updated past September 225

### DISCUSSION THREAD

**Discussion by uwagura** [NOAA-CEFI-Regional-Ocean-Modeling/seasonal-workflow#2] [General]
Now that this discussions page is setup, I figured I would make use of it :). 

ECMWF let me know that the GloFAS portal had been migrated from the Climate Data Store to the Early Warning Data Store, so this is the [new link](https://ewds.climate.copernicus.eu/datasets/cems-glofas-historical?tab=overview) to download GloFAS Data. If you are using the cdsapi tool to download GloFAS data, you will have to change your download url from `https://cds.climate.copernicus.eu/api` to `https://ewds.climate.copernicus.eu/api`

Additionally, they mentioned that: 

> GloFAS 3.1 is a legacy version replaced by version 4.0 on 26th July 2023. The systems were run in parallel until the 25th of September 2024 to enable users to migrate their workflows. There are no plans to produce data following the 3.1 configuration for further dates. The data portal reflects all that's available for this dataset. Please check the changes introduced in version 4.0 in this dedicated wiki page  https://confluence.ecmwf.int/display/CEMS/GloFAS+v4.0

@andrew-c-ross , looks we may need to use GloFAS version 4.0 for any data after September 23rd. Do we need to redownload anything from before this date? Let me know what you think

**Comment by andrew-c-ross** [comment]
Thanks for catching this, @uwagura . Long-term we probably want to switch to using GloFAS version 4 for the entire 1993--present, but for the near term we can use it for only the most recent time period. I think it would be cleanest and easiest if we switched at the start of a new year, so if it's available now we could start with downloading January 1 2024 to the present.

**Reply by uwagura** [reply]
Sounds good. Also, it looks like GloFAS v4 is at a higher resolution ( .05 degrees ) than version 3.1. Will this cause any issues?

**Reply by andrew-c-ross** [reply]
We'll have to see if the size of the data causes any slowness when interpolating it, but otherwise for NWA I do all of the modifications to the data after it has been interpolated onto the MOM6 grid. There should be just a few extremely minor changes needed in the code for creating the source grid for the interpolation.

We also need the static Local Drain Direction data, which according to the [Wiki](https://confluence.ecmwf.int/display/CEMS/Auxiliary+Data) is [here](https://confluence.ecmwf.int/download/attachments/242067380/ldd_glofas_v4_0.nc?version=1&modificationDate=1669994937993&api=v2). With the first chunk of GloFAs version 4 that you get, it would be good to check that the data have the same shape and coordinate values as this LDD map.

**Comment by yichengt900** [comment]
Thank you, @uwagura. @andrew-c-ross, @uwagura is currently focused on data preparation, and we will notify you once the data is ready. In the meantime, I am curious about the current status of the physic+BGC seasonal prediction. I see the [configuration file](https://github.com/NOAA-CEFI-Regional-Ocean-Modeling/seasonal-workflow/blob/main/config_nwa12_cobalt.yaml) and [post-processing scripts](https://github.com/NOAA-CEFI-Regional-Ocean-Modeling/seasonal-workflow/blob/main/postprocess_extract_fields.py) for the coupled run, but I cannot locate the example XML or template for the coupled case in the [XML](https://github.com/NOAA-CEFI-Regional-Ocean-Modeling/seasonal-workflow/tree/main/xml) folder. Is it possible that I may have overlooked something?

**Reply by andrew-c-ross** [reply]
I've already set up and run the entire forecast ensemble, but I'm behind on committing and pushing everything to this repository. Will do that soon. Took some work to figure out how to deal with the large BGC data.
